test/rubygems/test_require.rb

```diff
@@ -529,6 +529,65 @@ def test_default_gem_prerelease
     assert_equal %w[default-3.0.0.rc2], loaded_spec_names
   end
 
+  def test_default_gem_with_unresolved_gems_depending_on_it
```

Comment:
Can you show a real example? My understanding is that non default gems cannot shadow default gems (unless they are put in front of the `$LOAD_PATH` manually), and default gems take priority?

Reply:
Oh, maybe you're precisely suggesting that with this change the default gem may start not getting activated, in favor of the non default one?

Reply:
Oh, or maybe you mean when a higher non default version of the same gem needs to shadow the default version of that gem. In that case, we already have tests for that. See https://github.com/rubygems/rubygems/blob/271c9226589baacebfa18da939c05b4ae4f8a434/test/rubygems/test_require.rb#L481-L491.

Reply:
> Oh, maybe you're precisely suggesting that with this change the default gem may start not getting activated, in favor of the non default one?

potentially?

> My understanding is that non default gems cannot shadow default gems

do we have tests covering that behavior? if not, im worried it might have changed

Reply:
Default gems are already in the default LOAD_PATH while normal gems are not and require explicit activation. I think that's normally done by rescuing `LoadError` from standard `require`. I think this PR should not regress that.

I'm adding a new test to make sure 👍.